packages/vm/tests/api/consensus.spec.ts

```diff
@@ -0,0 +1,41 @@
+        const block = Block.fromRLPSerializedBlock(RLP, { common })
+      
+        const stateDB = level('/Users/jochem/Library/Ethereum/ethereumjs/rinkeby/state')
+        const trie = new Trie(stateDB)
+        const stateManager = new DefaultStateManager({ trie, common })
+        // Ensure we run on the right root
+        stateManager.setStateRoot(Buffer.from('135fae6d4909a072d77707a797769afc72f6239a72173e0f411c7700af66ac26', 'hex'))
```

Comment:
If I change this to:

```typescript
        // Ensure we run on the right root
        stateManager.setStateRoot(Buffer.from('0c8c521bd9c5272533db5ec1c9f73a6c727e343005f8a5c61e197d2953c0e165', 'hex'))
```
        
(This is the state root of block 14181). Then the test passes... (Valid receipt trie, valid root, etc.) There seems to be something going wrong with setting the state root somewhere? It seems to set the state root wrong at some point? My gut says that this might have to do with a bug in clique somewhere? @holgerd77

Reply:
I can't follow the argumentation, if you set the state root to the wrong state root (or am I missing something here?) then the test passes, does this say anything (except that it is *really* the state root that is wrong?)? 🤔  This can still be anything *leading* to a state root failure.

Please correct me if I am missing the point.

Reply:
This 0x0c8c... is the state root of block [14181](https://rinkeby.etherscan.io/block/14181). I don't know where this original 0x135f... root comes from? I also forgot if I copied some of your code since I cannot find the original (😅 )

Reply:
Update: ah, sorry, this is the pre-state root, right? Is this ensured that this is the correct one?

Reply:
Yes, it should be; it is the state root of the previous block (which is the state root *after* processing the block). I took the state root from #1091.

Reply:
Ah, I think my state root request for the script from [here](https://github.com/ethereumjs/ethereumjs-monorepo/pull/1091/files#diff-b16a7096add4d1d5cf968dd3ba950f16028c1634efa5b49f6bd652bbd9f5aacfR44) is likely wrong? But with the corrected state root the tests pass? Why this? 😄  Shouldn't they fail?

Reply:
Lunch time 🍜

Reply:
(or is it? I am getting confused, too hungry. 😛  )

Reply:
Yes good point - it should fail, since it fails in client. So that implies that client or clique is somehow messing up the state root.

Reply:
Also put a comment on your code above.